script.js

```diff
@@ -0,0 +1,108 @@
+const inputText = document.querySelector('#todo-input');
+const addTodoBtn = document.querySelector('#add-todo-btn');
+const todoListTitle = document.querySelector('#todo-list-title');
+const doneListTitle = document.querySelector('#done-list-title');
+const todoList = document.querySelector('.todo-list');
+const doneList = document.querySelector('.done-list');
+
+let items = [];
+
+// 입력한 할 일을 items에 추가
+const addNewTodo = () => {
+  const todoObject = {
+    id: null,
+    text: inputText.value,
+    isDone: false,
+  };
+
+  // 공백 입력, 중복 입력 방지
+  const index = items.findIndex((todo) => todo.text == inputText.value);
+  if (inputText.value && index === -1) {
+    items.push(todoObject);
+    render();
+  }
+
+  // 할 일 추가 후 인풋 값 비우기
+  inputText.value = '';
+};
+
+addTodoBtn.addEventListener('click', addNewTodo);
+
+// 엔터를 눌러도 입력되도록
+const enterKey = () => {
+  if (window.event.keyCode == 13) {
+    addNewTodo();
+  }
+};
+
+// 삭제 버튼을 누르면 해당하는 항목 삭제
+const deleteTodo = (e) => {
+  items = items.filter((todo) => 'bin' + todo.id !== e.target.id);
+  render();
+};
+
+// 할 일의 isDone을 토글
+const toggleTodo = (e) => {
+  const todo = items.find((todo) => 'todo' + todo.id === e.target.id);
+  todo.isDone = !todo.isDone;
+
+  render();
+};
+
+// 할 일과 완료한 일의 개수 세기
+const countTodo = () => {
+  const doneTodoCnt = items.filter((todo) => todo.isDone).length;
+  const todoCnt = items.length - doneTodoCnt;
+  todoListTitle.innerHTML = `📋 TO DO (${todoCnt})`;
+  doneListTitle.innerHTML = `💿 DONE (${doneTodoCnt})`;
+};
+
+// 할 일 목록을 화면에 렌더링
+const render = () => {
+  // 렌더링 전 todoList와 doneList 비우기
+  todoList.innerHTML = '';
+  doneList.innerHTML = '';
+
+  // 리스트로 만들어 보여주기
+  items.map((todo, index) => {
+    // id 값 부여
+    todo.id = index;
+
+    // 할 일, 완료한 일 개수 세기
+    countTodo();
+
+    // 리스트 만들기
+    const todoListItem = document.createElement('li');
+    const todoListItemText = document.createElement('span');
+    todoListItemText.setAttribute('id', 'todo' + index);
+    todoListItemText.textContent = todo.text;
+    todoListItemText.addEventListener('click', toggleTodo);
+    todoListItem.appendChild(todoListItemText);
+
+    // 삭제 버튼 만들어서 붙이기
+    const deleteBtn = document.createElement('i');
+    deleteBtn.setAttribute('class', 'delete-btn');
+    deleteBtn.setAttribute('id', 'bin' + index);
+    deleteBtn.addEventListener('click', deleteTodo);
+
+    todoListItem.appendChild(deleteBtn);
+    todo.isDone
+      ? doneList.appendChild(todoListItem)
+      : todoList.appendChild(todoListItem);
+  });
```

Comment:
```suggestion
  items.forEach((todo, index) => {
    // id 값 부여
    todo.id = index;

    // 할 일, 완료한 일 개수 세기
    countTodo();

    // 리스트 만들기
    const todoListItem = document.createElement('li');
    const todoListItemText = document.createElement('span');
    todoListItemText.setAttribute('id', 'todo' + index);
    todoListItemText.textContent = todo.text;
    todoListItemText.addEventListener('click', toggleTodo);
    todoListItem.appendChild(todoListItemText);

    // 삭제 버튼 만들어서 붙이기
    const deleteBtn = document.createElement('i');
    deleteBtn.setAttribute('class', 'delete-btn');
    deleteBtn.setAttribute('id', 'bin' + index);
    deleteBtn.addEventListener('click', deleteTodo);

    todoListItem.appendChild(deleteBtn);
    todo.isDone
      ? doneList.appendChild(todoListItem)
      : todoList.appendChild(todoListItem);
  });
```

특별히 map함수의 반환값을 쓰지는 않으므로 `forEach` 메소드가 더 적절해 보입니다.

Reply:
헉 그러네요 수정하겠습니다!!